libcloud/compute/drivers/dimensiondata.py

```diff
@@ -0,0 +1,594 @@
+        """
+        Send the /myaccount API request to DimensionData cloud and parse the
+        'orgId' from the XML response object. We need the orgId to use most
+        of the other API functions
+        """
+        if self._orgId is None:
+            body = self.request('myaccount').object
+            self._orgId = findtext(body, 'orgId', DIRECTORY_NS)
+        return self._orgId
+
+
+class DimensionDataStatus(object):
+    """
+    DimensionData API pending operation status class
+        action, requestTime, username, numberOfSteps, updateTime,
+        step.name, step.number, step.percentComplete, failureReason,
+    """
+    def __init__(self, action=None, requestTime=None, userName=None,
+                 numberOfSteps=None, updateTime=None, step_name=None,
+                 step_number=None, step_percentComplete=None,
+                 failureReason=None):
+        self.action = action
+        self.requestTime = requestTime
+        self.userName = userName
+        self.numberOfSteps = numberOfSteps
+        self.updateTime = updateTime
+        self.step_name = step_name
+        self.step_number = step_number
+        self.step_percentComplete = step_percentComplete
+        self.failureReason = failureReason
+
+    def __repr__(self):
+        return (('<DimensionDataStatus: action=%s, requestTime=%s, '
+                 'userName=%s, numberOfSteps=%s, updateTime=%s, '
+                 'step_name=%s, step_number=%s, '
+                 'step_percentComplete=%s, failureReason=%s')
+                % (self.action, self.requestTime, self.userName,
+                   self.numberOfSteps, self.updateTime, self.step_name,
+                   self.step_number, self.step_percentComplete,
+                   self.failureReason))
+
+
+class DimensionDataNetwork(object):
+    """
+    DimensionData network with location.
+    """
+
+    def __init__(self, id, name, description, location, privateNet,
+                 multicast, status):
+        self.id = str(id)
+        self.name = name
+        self.description = description
+        self.location = location
+        self.privateNet = privateNet
+        self.multicast = multicast
+        self.status = status
+
+    def __repr__(self):
+        return (('<DimensionDataNetwork: id=%s, name=%s, description=%s, '
+                 'location=%s, privateNet=%s, multicast=%s>')
+                % (self.id, self.name, self.description, self.location,
+                   self.privateNet, self.multicast))
+
+
+class DimensionDataNodeDriver(NodeDriver):
+    """
+    DimensionData node driver.
+    """
+
+    connectionCls = DimensionDataConnection
+    name = 'DimensionData'
+    website = 'http://www.dimensiondata.com/'
+    type = Provider.DIMENSIONDATA
+    features = {'create_node': ['password']}
+
+    def create_node(self, **kwargs):
+        """
+        Create a new DimensionData node
+
+        :keyword    name:   String with a name for this new node (required)
```

Comment:
Can we please also fix this and explicitly declare arguments in the method signature?

https://libcloud.readthedocs.org/en/latest/development.html#don-t-abuse-kwargs